src/tests/engines_util/test_case.cpp

```diff
@@ -43,14 +67,13 @@ compare_values(const ov::Tensor& expected, const ov::Tensor& result, const size_
 template <typename T>
 typename std::enable_if<std::is_class<T>::value, testing::AssertionResult>::type
 compare_values(const ov::Tensor& expected_tensor, const ov::Tensor& result_tensor, const size_t tolerance_bits) {
-    auto exp_host_t = std::make_shared<ngraph::HostTensor>(expected_tensor.get_element_type(),
-                                                           expected_tensor.get_shape(),
-                                                           expected_tensor.data());
-    auto res_host_t = std::make_shared<ngraph::HostTensor>(result_tensor.get_element_type(),
-                                                           result_tensor.get_shape(),
-                                                           result_tensor.data());
-    const auto expected = read_vector<T>(exp_host_t);
-    const auto result = read_vector<T>(res_host_t);
+    std::vector<T> expected(expected_tensor.get_size());
```

Comment:
Why do you need to copy these tensors? I think you can extend all_close_f to compare the original memory. Or copy from tensor to vector with double. Right now you have 2 copies

Reply:
Previously we also did have a copy with "read_tensor".
Ok, I will change it

Reply:
Previously we also did have a copy with "read_tensor".
Ok, I will change it

Reply:
Previously we also did have a copy with "read_tensor".
Ok, I will change it

Reply:
all_close_f can compare only double and float values. Here we use f16 and bf16. So we need to perform a conversion.
I changed it to have only 1 copy with conversion.